calculator/main.py

```diff
@@ -26,8 +26,18 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
-
+    # perform operation
+    for i in params:
+        if not isinstance(i, (int, float)):
+            raise InvalidParams("Given params are invalid.")
+    total = calc['operations'][operation](*params)
+    
+    # get the execution date-time and append in the history list
+    # double brackets since 'append' takes only one argument
+    calc['history'].append((str(datetime.now()), operation, params, total))
```

Comment:
huh. didn't realise that the default for `datetime.now()` is the format given in the test @jsymons we could've avoided some work but on the bright side we learnt about `strftime` and its formatting parameters